### ISSUE TITLE
What is the control and data flow among parts?

### ISSUE THREAD

**Issue by Derevi** [Derevi/breakfast-club-as1#4]

**Comment by jnuguid1** [comment]
I will do this part

**Comment by jnuguid1** [comment]
[Report 1 - Control and Data Flow.docx](https://github.com/Derevi/breakfast-club-as1/files/7271972/Report.1.-.Control.and.Data.Flow.docx)

**Comment by Derevi** [comment]
- If the buffer manager, handles buffer related things, is there some component or service that handles stuff that is on storage. Like if the buffer manager cant find something on its buffer pool, does control get passed to some kind of "storage manager" that gets some page on storage and forwards the data and control back to the buffer manager? 
- Maybe a little more detail would be nice, about some of the control. Like you mention that client sends query to server, server process, and sends results, maybe a few more steps in between, without overlapping to much on the other sections like  issue #7 maybe you could say something like (this is a super rough guess btw..): 
client sends query data -> 
server receives daata and control ->
control and query msg data sent to query preprocessing subsystem -> 
command data and control sent to executor -> 
executor executes and request data from buffer storage, so hands control to it ->
buffer/storage stuff , as mentioned in your report-> 
control and raw data passed to some rendering system maybe??->
return data and control back to client, 
- There seems to be some stuff closer to implementation level, like when you talk about the bytes in a message, from the assignment handout, interpret it how you like:
_"You are not to concentrate on minor components, such as classes and procedures, which are smaller than
packages or modules. However, you may wish to discuss important abstractions, patterns, classes, data
structures or algorithms that are critical to the success of the architecture. "_
- Note on discrepancy: (see pg 2180 in the documentation, US layout version).  
The postgres server is the entire postgres application itself. The "postgres server" is not the same as the "postgres server process". Postgres server uses an actual server process in order to communicate with the client but this is not the "postgres server" it is the server process that runs within the postgres server. 
postgres backend should be referring to the actual postgres server application instance that is running, while postgres server process is the component that postgres server uses in order to communicate with clients.
in short:
postgres server = postgres backend + postgres server process
postgres server process = thing that allows client to communicate with postgres server
